rclcpp/src/rclcpp/parameter_client.cpp

```diff
@@ -144,6 +144,41 @@ AsyncParametersClient::get_parameters(
   return future_result;
 }
 
+std::shared_future<std::vector<rcl_interfaces::msg::ParameterDescriptor>>
+AsyncParametersClient::describe_parameters(
+  const std::vector<std::string> & names,
+  std::function<
+    void(std::shared_future<std::vector<rcl_interfaces::msg::ParameterDescriptor>>)
+  > callback)
+{
+  auto promise_result =
+    std::make_shared<std::promise<std::vector<rcl_interfaces::msg::ParameterDescriptor>>>();
+  auto future_result = promise_result->get_future().share();
+
+  auto request = std::make_shared<rcl_interfaces::srv::DescribeParameters::Request>();
+  request->names = names;
+
+  describe_parameters_client_->async_send_request(
+    request,
+    [promise_result, future_result, callback](
+      rclcpp::Client<rcl_interfaces::srv::DescribeParameters>::SharedFuture cb_f)
+    {
+      std::vector<rcl_interfaces::msg::ParameterDescriptor> descriptors;
+      auto & descs = cb_f.get()->descriptors;
+      for (auto & desc : descs) {
+        descriptors.push_back(
+          static_cast<rcl_interfaces::msg::ParameterDescriptor>(desc));
```

Comment:
Can we not use `auto` here?  I know that we use it elsewhere in this file, but it is really hard for me to tell what the types of all of this is (and hence why the `static_cast` is needed).  This would be much easier to read if we used the real types.